### PR TITLE
More robust faults on BMS

### PR DESCRIPTION
**Pull request by gtaharaedmonds**
### Changelist 

Made faults on BMS more robust by adding debounce times. This should make the faults resistance against random noise or anything else leading to intermittent bad measurements. Currently the values are 500ms for voltages, 1000ms for temperatures, and 100ms for TS current. Also, added descriptions for all alerts.

Also, made open wire check a warning so its now up to the driver whether or not we want to shut down.

Also also, 45C overtemp condition during charging is now based on `io_charger_isChargerConnected` instead of just checking current. This means we'll keep using the 60C value when driving with regen.

### Testing Done

Updated unit tests.

### CODE REVIEW

File: firmware/quadruna/BMS/src/app/app_accumulator.c [hunk 1]
```diff
@@ -461,36 +474,45 @@ bool app_accumulator_checkFaults(void)
     float min_allowable_cell_temp = MIN_CELL_DISCHARGE_TEMP_DEGC;
 
     // if we are charging, max cell temp is 45C not 60C
-    if (app_tractiveSystem_getCurrent() > 3.0f)
+    if (io_charger_isConnected())
```
Comment (will-chaba) [inline firmware/quadruna/BMS/src/app/app_accumulator.c:477]:
```suggestion
io_charger_isConnected pin does not exist on Quadruna, use CAN message instead
    if (app_canRx_BRUSA_IsConnected_get();)
   
```

File: can_bus/quadruna/BMS/BMS_alerts.json [hunk 2]
```diff
@@ -6,113 +6,113 @@
...
         },
         "Charger": {
             "id" :160,
-            "description" : "Example"
+            "description" : "TODO @Will"
         },
         "ChargerDisconnectedDuringCharge": {
             "id" :161,
-            "description" : "Example"
+            "description" : "TODO @Will"
```
Comment (will-chaba) [inline can_bus/quadruna/BMS/BMS_alerts.json:103]:
```suggestion
            "description" : "Charger was disconnected while in charge state"
```

File: can_bus/quadruna/BMS/BMS_alerts.json [hunk 3]
```diff
@@ -6,113 +6,113 @@
...
         },
         "CellOvertemp": {
             "id" :159,
-            "description" : "Example"
+            "description" : "Cell is critically undervoltage (>45degC if charging, >60degC if not)."
         },
         "Charger": {
             "id" :160,
-            "description" : "Example"
+            "description" : "TODO @Will"
```
Comment (will-chaba) [inline can_bus/quadruna/BMS/BMS_alerts.json:99]: This and "ChargerExternalShutdown" seem to be redundant to me, maybe remove this message and just use the other?

File: can_bus/quadruna/BMS/BMS_alerts.json [hunk 4]
```diff
@@ -6,113 +6,113 @@
...
         },
         "CellUndervoltage": {
             "id" :155,
-            "description" : "Example"
+            "description" : "Cell is critically undervoltage (<3.0V)"
         },
         "CellOvervoltage": {
             "id" :156,
-            "description" : "Example"
+            "description" : "Cell is critically overvoltage (<4.2V)"
```
Comment (will-chaba) [inline can_bus/quadruna/BMS/BMS_alerts.json:83]:
```suggestion
            "description" : "Cell is critically overvoltage (>4.2V)"
```

**Review by will-chaba — CHANGES_REQUESTED** [review]
Some small fixes needed

File: can_bus/quadruna/BMS/BMS_alerts.json (hunk 3 above)
Comment (gtaharaedmonds) [reply can_bus/quadruna/BMS/BMS_alerts.json:99]: they are different conditions though right? probably useful to specify which one it is?

File: firmware/quadruna/BMS/src/app/app_tractiveSystem.h [hunk 5]
```diff
@@ -7,8 +7,24 @@
...
+// Taken from our cell's datasheet on Confluence:
+// https://ubcformulaelectric.atlassian.net/wiki/pages/viewpageattachments.action?pageId=55214081&preview=%2F55214081%2F55214090%2FSLPB9742126-3.7V-5900mAh-10C-V2018.pdf
+#define MAX_TS_DISCHARGE_CURRENT_PER_CELL_AMPS (-88.5f)
+#define MAX_TS_CHARGE_CURRENT_PER_CELL_AMPS (23.6f)
+
+// As per datasheet, our cells can do the max discharge current for up to 3s and the peak charge current for 100ms. We
+// shouldn't be anywhere near the max discharge current during driving, so we can use 100ms for charge and discharge.
+#define TS_OVERCURRENT_DEBOUNCE_DURATION_MS (100U)
+
+#define CELLS_IN_PARALLEL (3)
```
[the remark this answers is not in the document]
Comment (gtaharaedmonds) [reply firmware/quadruna/BMS/src/app/app_tractiveSystem.h:20]: im feeling lazy

**Review by will-chaba — APPROVED** [review]
Looks good to me!

**Pull request merged**